### PR TITLE
doctest: include doctest.cmake in the package

### PR DESCRIPTION
**Pull request by p-podsiadly**
Specify library name and version:  **doctest/2.3.5**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

This change adds `doctest.cmake` script to the package. The script defines a `doctest_discrover_tests` function, which sets up CTest tests for test cases in an executable.

### CODE REVIEW

File: recipes/doctest/2.3.5/conanfile.py [hunk 1]
```diff
@@ -24,11 +24,22 @@ def package(self):
...
+
     def package_info(self):
         if self._is_mingw:
             # See https://sourceforge.net/p/mingw-w64/bugs/727/
             # can't use destructors in thread_local with mingw
             self.cpp_info.defines.append("DOCTEST_THREAD_LOCAL=")
 
+        self.cpp_info.builddirs.append("lib/cmake")
+        self.cpp_info.build_modules.append("lib/cmake/doctest.cmake")
+
```
Comment (uilianries) [inline recipes/doctest/2.3.5/conanfile.py]: Why did you add `doctestAddTests.cmake` if it's not listed in `self.cpp_info.build_modules` ?

Comment (p-podsiadly) [reply recipes/doctest/2.3.5/conanfile.py]: `doctestAddTests.cmake` is not meant to be included directly. `doctest.cmake` adds a `POST_BUILD` command, which calls it to discover tests (it's happening in [here](https://github.com/onqtam/doctest/blob/503de03ee2cbbfc58c2063dce36dc96f917f2da3/scripts/cmake/doctest.cmake#L138), `_DOCTEST_DISCOVER_TESTS_SCRIPT` points to the location of `doctestAddTests.cmake`).

Comment (uilianries) [reply recipes/doctest/2.3.5/conanfile.py]: thanks for answering, so we are good!

File: recipes/doctest/2.3.5/test_package/CMakeLists.txt [hunk 2]
```diff
@@ -3,9 +3,14 @@ project(test_package)
 
 include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
 conan_basic_setup()
+conan_include_build_modules()
```
Comment (danimtb) [inline recipes/doctest/2.3.5/test_package/CMakeLists.txt]: This is not needed, as `conan_basic_setup()` already includes the call to `conan_include_build_modules()`

Comment (p-podsiadly) [reply recipes/doctest/2.3.5/test_package/CMakeLists.txt]: Done.
